### PR TITLE
Fix builtin make integer type check

### PR DESCRIPTION
**Pull request by sjamesr**
Fixes https://github.com/go-lang-plugin-org/go-lang-idea-plugin/issues/2132.

### CODE REVIEW

**Comment by dlsniper** [comment]
@sjamesr does this work for:
```go
	var b interface{} = 1
	_ = make(chan int, b.(int))
```
as well?

**Comment by sjamesr** [comment]
@dlsniper I believe so, I just added a test for it

File: .idea/codeStyleSettings.xml [hunk 1]
```diff
@@ -133,7 +133,6 @@
           <option name="ALIGN_MULTILINE_THROWS_LIST" value="true" />
           <option name="ALIGN_MULTILINE_EXTENDS_LIST" value="true" />
           <option name="ALIGN_MULTILINE_PARENTHESIZED_EXPRESSION" value="true" />
-          <option name="ALIGN_SUBSEQUENT_SIMPLE_METHODS" value="true" />
```
Comment (zolotov) [inline .idea/codeStyleSettings.xml]: ?

Comment (sjamesr) [inline .idea/codeStyleSettings.xml]: reverted, sorry

File: src/com/goide/highlighting/GoAnnotator.java [hunk 2]
```diff
@@ -183,7 +186,8 @@ private static boolean isIntegerType(@Nullable GoType type) {
     if (type == null) return false;
     GoTypeReferenceExpression ref = type.getTypeReferenceExpression();
     if (ref == null) return false;
-    return GoPsiImplUtil.builtin(type) && INT_TYPE_NAMES.contains(ref.getText());
+    PsiElement resolve = ref.getReference().resolve();
+    return GoPsiImplUtil.builtin(resolve) && INT_TYPE_NAMES.contains(ref.getText());
```
Comment (zolotov) [inline src/com/goide/highlighting/GoAnnotator.java]: it makes sense to filter out references by name before resolving

Comment (sjamesr) [inline src/com/goide/highlighting/GoAnnotator.java]: Done

**Comment by zolotov** [comment]
Very fast response. Thanks :)

**Pull request merged**